test/test_runner.go

```diff
@@ -497,9 +509,22 @@ func contractValueHandler(
 		compositeType,
 	)
 
+	// In unit tests, contracts are imported with string locations, e.g
+	// import FooContract from "../contracts/FooContract.cdc"
+	if _, ok := compositeType.Location.(common.StringLocation); ok {
+		return stdlib.StandardLibraryValue{
+			Name:           declaration.Identifier.Identifier,
+			Type:           constructorType,
+			DocString:      declaration.DocString,
+			Kind:           declaration.DeclarationKind(),
+			Position:       &declaration.Identifier.Pos,
+			ArgumentLabels: constructorArgumentLabels,
+		}
+	}
+
 	return stdlib.StandardLibraryValue{
 		Name:           declaration.Identifier.Identifier,
-		Type:           constructorType,
+		Type:           compositeType,
```

Comment:
If the import is declared with the address, where would it load the contract instance from?

Reply:
Ah, that's a great catch Supun. I was little bit ahead of my self. This actually comes from the PR that allows importing deployed contracts in a test script (https://github.com/onflow/cadence-tools/pull/197).
However, I back-ported it here, to make this PR fully functional. I forgot to add the relevant test case though :sweat_smile: 
The general idea is that you import a contract from its deployed address on the emulator blockchain, to make the type-checking happy.
I've added the fix here: https://github.com/onflow/cadence-tools/pull/205/commits/fb17f60ba7b2cf536e24b8b5212d1d0e0e958dfb.
In a different PR, I will add the changes for using `flow.MonotonicEmulator`, so that addresses are sequential and know, for imports to not regress.

Reply:
Thanks! So in fb17f60ba7b2cf536e24b8b5212d1d0e0e958dfb, it only works if the contract `init` has no params, correct? Maybe just keep both paths to work in the same way. i.e: both return the constructor, so in the testing code, developers have to create a contract value by themselves before using it.

Reply:
Nice catch :clap: This actually has to work irrespective of the params count for `init`. I have updated this in https://github.com/onflow/cadence-tools/pull/205/commits/98ee3d8d27212c3280ed2c58369b5075bddb6a8d

Reply:
I left a comment above reg: the contract init.
To Keep things simple, could we maybe keep only the constructor for now? So developers can construct a value in the code, and then access types through that. That way, it will be unified for both scenarios (importing by path vs importing by address).
We can figure out/improve this again later, but the rest of the PR doesn't need to be blocked on that.

Reply:
@SupunS By unifying the `parseAndCheckImport` and removing entirely the `EmulatorBackend` that we had on `TestRunner`, this change is now necessary. Removing it breaks already existing features, e.g event retrieval:

```shell
--- FAIL: TestGetEventsFromIntegrationTests (1.59s)
    /home/mpeter/Dev/cadence-tools/test/test_framework_test.go:3992: 
        	Error Trace:	/home/mpeter/Dev/cadence-tools/test/test_framework_test.go:3992
        	Error:      	Received unexpected error:
        	            	Execution failed:
        	            	error: Checking failed:
        	            	error: value of type `((): MetadataViews)` has no member `NFTCollectionData`
        	            	   --> f8d6e0586b0a20c7.ExampleNFT:340:37
        	            	
        	            	error: value of type `((): ExampleNFT)` has no member `CollectionStoragePath`
        	            	   --> f8d6e0586b0a20c7.ExampleNFT:341:44
```

Reply:
Oh I see. Makes sense in that case. 👍 
Can you please add this explanation as a comment to that section?

Reply:
Added in https://github.com/onflow/cadence-tools/pull/205/commits/08d88bbb46f9bdedb9c6146252cf1b596a6171ed